solana-labs/wallet-adapter
# Enable strict ESM exports

This PR fixes #246 by causing all builds to produce .mjs files.

It doesn't address #241 yet, see https://github.com/solana-labs/wallet-adapter/pull/254#issuecomment-1008227270

It also removes the need to dynamically load the WalletProvider in Next's SSR, which will help with static page generation.

cc @cogoo @gpoole @stanleycyang

To test this, from the `strict-esm` branch:

```
yarn nuke
yarn
yarn build
```

This builds all the packages, but you'll get an error from the `react-ui-starter` package. This is because it's still using CRA v4, which doesn't handle the new imports.

However, if you then do this:

```
cd packages/starter/next-12-starter
yarn build
```

This builds now with no errors.

```
yarn dev
open http://localhost:3000
```

The page renders with no issues and the WalletAdapter buttons work.

We'll need to update the CRA v4 based starter projects to v5, which should fix this with no additional configuration.
